src/pudl/analysis/plant_parts_eia.py

```diff
@@ -0,0 +1,1733 @@
+3              1               d               2                  GT             75
+4              1               e               3                  GT             75
+
+>>> df_plant_part = pd.DataFrame({
+...     'plant_id_eia': [1, 1],
+...     'plant_part': ['plant_unit', 'plant_prime_mover'],
+...     'unit_id_pudl': [1, pd.NA],
+...     'prime_mover_code': [pd.NA, 'GT',],
+...     'capacity_mw': [200, 150],
+... })
+>>> df_plant_part
+    plant_id_eia           plant_part    unit_id_pudl    prime_mover_code    capacity_mw
+0              1           plant_unit               1                <NA>            200
+1              1    plant_prime_mover            <NA>                  GT            150
+
+In this case last, the ``plant_unit`` record would have a null
+``plant_prime_mover`` because the unit contains more than one
+``prime_mover_code``. Same goes for the ``unit_id_pudl`` of the
+``plant_prime_mover``. This is handled in the :class:``AddConsistentAttributes``.
+
+**Overview of flow for generating the master unit list:**
+
+The three main classes which enable the generation of the plant-part table are:
+
+* :class:`MakeMegaGenTbl`: All of the plant parts are compiled from generators.
+  So this class generates a big dataframe of generators with any ID and data
+  columns we'll need. This is also where we add records regarding utility
+  ownership slices. The table includes two records for every generator-owner:
+  one for the "total" generator (assuming the owner owns 100% of the generator)
+  and one for the report ownership fraction of that generator with all of the
+  data columns scaled to the ownership fraction.
+* :class:`LabelTrueGranularities`: This class creates labels for all generators
+  which note wether the plant-part records that will be compiled from each
+  generator will be a "true granulary", as described above.
+* :class:`MakePlantParts`: This class uses the generator dataframe, the
+  granularity dataframe from the above two classes as well as the information
+  stored in :py:const:`PLANT_PARTS` to know how to aggregate each of the plant
+  parts. Then we have plant part dataframes with the columns which identify the
+  plant part and all of the data columns aggregated to the level of the plant
+  part. With that compiled plant part dataframe we also add in qualifier
+  columns with :class:`AddConsistentAttributes`. A qualifer column is a column which
+  contain data that is not endemic to the plant part record (it is not one of
+  the identifying columns or aggregated data columns) but the data is still
+  useful data that is attributable to each of the plant part records. For more
+  detail on what a qualifier column is, see :meth:`AddConsistentAttributes.execute`.
+
+Lines needed to generate the plant-parts df:
+
+``
+import pudl
+# make the pudl_out object
+pudl_engine = sa.create_engine(pudl.workspace.setup.get_defaults()['pudl_db'])
+pudl_out = pudl.output.pudltabl.PudlTabl(pudl_engine,freq='AS')
+``
```

Comment:
These code blocks are not rendering on the [docs](https://catalystcoop-pudl--1157.org.readthedocs.build/en/1157/autoapi/pudl/analysis/plant_parts_eia/index.html). Here is [some documentation](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/code_blocks.html#python) that describes how to add code blocks on read the docs.

Reply:
ty!!